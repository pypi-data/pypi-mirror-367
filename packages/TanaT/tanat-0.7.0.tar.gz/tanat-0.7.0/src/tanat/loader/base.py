#!/usr/bin/env python3
"""
Loader base class
"""

from abc import ABC, abstractmethod

from pypassist.mixin.cachable import Cachable
from pypassist.mixin.settings import SettingsMixin
from pypassist.mixin.registrable import Registrable, UnregisteredTypeError
from pydantic_core import core_schema

from .exception import UnregisteredLoaderTypeError


class Loader(
    ABC,
    Registrable,
    Cachable,
    SettingsMixin,
):
    """Base class for all loaders"""

    _REGISTER = {}
    _TYPE_SUBMODULE = "type"

    def __init__(self, settings):
        """
        Initialize the loader with the given settings.

        Args:
            settings: Configuration settings for the loader.

        Raises:
            ValueError: If the settings type is invalid.
        """
        SettingsMixin.__init__(self, settings)
        Cachable.__init__(self)

    def load(self):
        """
        Load data using the settings provided at initialization.

        Returns:
            The loaded data.
        """
        return self._load_impl(self.settings)

    @abstractmethod
    def _load_impl(self, settings):
        """
        Implementation of the data loading logic.

        Args:
            settings: Configuration settings for the loader.
            Mostly there for caching invalidation

        Returns:
            The loaded data.
        """

    def __call__(self):
        """
        Allow the loader to be called as a function.

        Returns:
            The loaded data.
        """
        return self.load()

    @classmethod
    def get_loader(cls, ltype, settings=None):
        """
        Retrieve and instantiate a Loader.

        Args:
            ltype: Type of loader to use, resolved via type registry.
            settings: Loader-specific settings dictionary or dataclass.

        Returns:
            An instance of the Loader configured
            with the provided settings.

        Raises:
            UnregisteredLoaderTypeError: If the loader type is not registered.
        """
        try:
            loader_cls = cls.get_registered(ltype)
            return loader_cls(settings=settings)
        except UnregisteredTypeError as err:
            raise UnregisteredLoaderTypeError(
                f"Unknown loader type: '{ltype}'."
                f"Available loaders: {cls.list_registered()}"
            ) from err

    def refresh(self):
        """
        Reload the data.

        Returns:
            The reloaded data.
        """
        self.clear_cache()
        return self.load()

    def __get_pydantic_core_schema__(self, handler):  # pylint: disable=unused-argument
        """Pydantic schema override."""
        return core_schema.any_schema()
