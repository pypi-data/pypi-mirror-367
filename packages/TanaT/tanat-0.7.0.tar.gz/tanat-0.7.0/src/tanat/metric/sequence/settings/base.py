#!/usr/bin/env python3
"""
Base settings for sequence metrics.
"""

from pydantic.dataclasses import dataclass

from .parallel import ParallelSettings


@dataclass
class BaseSequenceMetricSettings(ParallelSettings):
    """
    Base class for sequence metric settings.
    """
