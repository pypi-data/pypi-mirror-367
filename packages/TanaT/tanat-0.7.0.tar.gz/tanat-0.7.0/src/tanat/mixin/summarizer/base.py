#!/usr/bin/env python3
"""
Summarizer utilities.
"""

from abc import abstractmethod
import logging
from pathlib import Path

from pypassist.utils.export import export_string

LOGGER = logging.getLogger(__name__)


class BaseSummarizerMixin:
    """Base class for generating summaries."""

    # Constantes de formatage centralisées
    SECTION_WIDTH = 50
    BORDER_CHAR = "─"
    INDENT = "  "

    @property
    def statistics(self):
        """Generate statistics section."""
        return self._format_statistics()

    @property
    def summary(self):
        """Generate a formatted summary and return it."""
        return self.summarize()

    def summarize(self, export_to=None, *, filename="summary.txt"):
        """
        Generate a formatted summary with optional file export.

        Args:
            export_to (str): Path to export the summary to.
            filename (str): Name of the file to export the summary to.

        Returns:
            str or None: The formatted summary text (or None if exported).
        """
        sections = [
            self._format_header(),
            self._format_statistics(),
            self._format_data_preview(),
        ]

        summary_text = "\n".join(filter(None, sections))

        if export_to:
            full_path = Path(export_to) / filename
            export_string(summary_text, full_path)
            return None
        return summary_text

    def _format_header(self):
        """Generate a clean, consistent header."""
        title = self._get_title()
        border = self.BORDER_CHAR * self.SECTION_WIDTH
        return f"┌{border}┐\n│ {title:<{self.SECTION_WIDTH-2}} │\n└{border}┘"

    def _get_title(self):
        """Get the summary title."""
        return f"{self.__class__.__name__} summary"

    def _format_section(self, title, content):
        """Format a section with consistent styling."""
        if not content:
            return ""
        separator = self.BORDER_CHAR * (self.SECTION_WIDTH // 2)
        return f"\n{title}\n{separator}\n{content}"

    def _format_stat_line(self, label, value):
        """Format a single statistics line."""
        # total_width = self.SECTION_WIDTH // 2 - len(self.INDENT)
        width = (self.SECTION_WIDTH - len(self.INDENT)) // 2
        label_width = width - 5
        return f"{self.INDENT}{label:<{label_width}}{value:<15}"

    def _indent_text(self, text, level=1):
        """Indent text consistently."""
        indent = self.INDENT * level
        return "\n".join(f"{indent}{line}" for line in text.splitlines())

    @abstractmethod
    def _format_statistics(self):
        """Generate statistics section. Must be implemented by child classes."""

    @abstractmethod
    def _format_data_preview(self):
        """Generate data preview section. Must be implemented by child classes."""
