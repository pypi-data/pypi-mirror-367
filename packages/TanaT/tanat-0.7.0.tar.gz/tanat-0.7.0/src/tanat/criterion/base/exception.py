#!/usr/bin/env python3
"""
Criterion exceptions.
"""

from ...exception import TanatException


class CriterionException(TanatException):
    """
    Base exception for all criterion-related errors in the TanaT framework.
    """


class InvalidCriterionError(CriterionException):
    """
    Raised when an invalid or improperly configured criterion is encountered.
    """
