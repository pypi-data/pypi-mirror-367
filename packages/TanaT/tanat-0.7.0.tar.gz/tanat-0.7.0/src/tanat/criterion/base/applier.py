#!/usr/bin/env python3
"""
Base class for criterion applier.
"""

from abc import ABC, abstractmethod

from pypassist.mixin.registrable import Registrable, UnregisteredTypeError
from pypassist.mixin.settings import SettingsMixin
from pydantic_core import core_schema

from .settings import Criterion
from ...sequence.base.sequence import Sequence
from ...sequence.base.pool import SequencePool
from ...trajectory.pool import TrajectoryPool
from ...trajectory.trajectory import Trajectory


class CriterionApplier(ABC, Registrable, SettingsMixin):
    """
    Abstract base class for all criterion appliers in the TanaT framework.

    Provides a standardized interface for initializing and managing
    criterion-based filtering across different data types and levels.
    """

    def __init__(self, settings):
        """
        Initialize the CriterionApplier with specific settings.

        Args:
            settings: Criterion object defining filtering parameters.
        """
        SettingsMixin.__init__(self, settings)

    @classmethod
    def init(cls, criterion, criterion_type=None):
        """
        Initialize a Criterion applier class dynamically.

        Supports initialization from dictionary or Criterion object.

        Args:
            criterion: Criterion settings (dict or Criterion object)
            criterion_type: Type of criterion for registry resolution

        Returns:
            Configured Criterion applier instance

        Raises:
            ValueError: If initialization parameters are invalid
        """
        if isinstance(criterion, dict):
            if criterion_type is None:
                raise ValueError(
                    "criterion_type must be provided if settings is a dictionary."
                )

        else:
            if not isinstance(criterion, Criterion):
                raise ValueError(
                    f"settings must be a Criterion object or a dictionary. "
                    f"Got {criterion!r}."
                )
            # pylint: disable=protected-access
            criterion_type = criterion._REGISTER_NAME

        try:
            criterion = cls.get_registered(criterion_type)(settings=criterion)
        except UnregisteredTypeError as err:
            cls._unregistered_criterion_error(criterion_type, err)

        return criterion

    def _is_sequence_instance(self, sequence):
        """
        Check if the given object is a Sequence instance.

        Args:
            sequence: Object to check

        Returns:
            bool: True if object is a Sequence, False otherwise
        """
        return isinstance(sequence, Sequence)

    def _is_sequence_pool_instance(self, sequence_pool):
        """
        Check if the given object is a SequencePool instance.

        Args:
            sequence_pool: Object to check

        Returns:
            bool: True if object is a SequencePool, False otherwise
        """
        return isinstance(sequence_pool, SequencePool)

    def _is_trajectory_pool_instance(self, trajectory_pool):
        """
        Check if the given object is a TrajectoryPool instance.

        Args:
            trajectory_pool: Object to check

        Returns:
            bool: True if object is a TrajectoryPool, False otherwise
        """
        return isinstance(trajectory_pool, TrajectoryPool)

    def _is_trajectory_instance(self, trajectory):
        """
        Check if the given object is a Trajectory instance.

        Args:
            trajectory: Object to check

        Returns:
            bool: True if object is a Trajectory, False otherwise
        """
        return isinstance(trajectory, Trajectory)

    @classmethod
    @abstractmethod
    def _unregistered_criterion_error(cls, criterion_type, err):
        """
        Raise a custom error for unregistered criterion types.

        Args:
            criterion_type: Type of unregistered criterion
            err: Original UnregisteredTypeError

        Raises:
            Custom exception specific to the criterion context
        """

    def __get_pydantic_core_schema__(self, handler):  # pylint: disable=unused-argument
        """
        Provide a custom Pydantic core schema for validation.

        Returns:
            Pydantic core schema for this class
        """
        return core_schema.any_schema()
