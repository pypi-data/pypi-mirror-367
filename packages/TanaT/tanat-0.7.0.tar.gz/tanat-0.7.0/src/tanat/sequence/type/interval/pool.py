#!/usr/bin/env python3
"""
Interval sequence pool class.
"""

from ...base.pool import SequencePool
from .sequence import IntervalSequence
from .settings import IntervalSequenceSettings


class IntervalSequencePool(SequencePool, register_name="interval"):
    """
    Sequence pool for interval sequences.
    """

    SETTINGS_DATACLASS = IntervalSequenceSettings

    def _create_sequence_instance(
        self, id_value, sequence_data, settings, metadata, static_data
    ):
        """Create IntervalSequence instance."""
        return IntervalSequence(
            id_value,
            sequence_data,
            settings,
            metadata,
            static_data,
        )
