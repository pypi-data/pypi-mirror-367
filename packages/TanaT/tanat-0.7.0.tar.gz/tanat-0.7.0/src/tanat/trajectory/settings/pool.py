#!/usr/bin/env python3
"""
Settings for TrajectoryPool and its factory.
"""

from pydantic.dataclasses import dataclass
from pypassist.dataclass.decorators.viewer.decorator import viewer

from .base import BaseTrajectorySettings


@viewer
@dataclass
class TrajectoryPoolSettings(BaseTrajectorySettings):
    """
    Settings for TrajectoryPool.

    Attributes:
        id_column: The name of the column representing the ID.
        static_features: The names of the columns representing the static features.
        intersection: If True, uses the intersection of IDs across SequencePools.
    """

    intersection: bool = False
