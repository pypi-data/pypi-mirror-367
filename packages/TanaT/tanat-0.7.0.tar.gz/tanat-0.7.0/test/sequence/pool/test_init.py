#!/usr/bin/env python3
"""
Test initialization of sequence pools.
"""

import pytest

import pandas as pd

from tanat.sequence.base.pool import SequencePool


class TestInitSequencePool:
    """
    Test initialization of sequence pools.
    """

    @pytest.mark.parametrize(
        "seq_type,settings",
        [
            ## -- event
            (
                "event",
                {
                    "id_column": "patient_id",
                    "time_column": "date",
                    "entity_features": ["event_type", "provider"],
                    "static_features": [
                        "gender",
                        "age",
                        "insurance",
                        "chronic_condition",
                    ],
                },
            ),
            ## -- state
            (
                "state",
                {
                    "id_column": "patient_id",
                    "entity_features": ["health_state", "condition"],
                    "start_column": "start_date",
                    "end_column": "end_date",
                    "static_features": [
                        "gender",
                        "age",
                        "insurance",
                        "chronic_condition",
                    ],
                },
            ),
            ## -- interval
            (
                "interval",
                {
                    "id_column": "patient_id",
                    "entity_features": [
                        "medication",
                        "administration_route",
                        "dosage",
                    ],
                    "start_column": "start_date",
                    "end_column": "end_date",
                    "static_features": [
                        "gender",
                        "age",
                        "insurance",
                        "chronic_condition",
                    ],
                },
            ),
        ],
    )
    def test_init_sequence_pool(self, pool_data, seq_type, settings):
        """
        Test initialization of sequence pools.
        """
        sequence_data = pool_data[seq_type]
        static_data = pool_data["static_data"]

        seqpool = SequencePool.init(
            seq_type, sequence_data, settings=settings, static_data=static_data
        )

        assert isinstance(seqpool, SequencePool)
        assert isinstance(seqpool.sequence_data, pd.DataFrame)
        assert isinstance(seqpool.static_data, pd.DataFrame)

    @pytest.mark.parametrize("anchor", ["start", "middle", "end"])
    def test_anchoring_interval(self, pool_data, anchor, snapshot):
        """
        Test anchoring behavior for interval sequence types.
        Should modify entity order.
        """
        settings = {
            "id_column": "patient_id",
            "entity_features": [
                "medication",
                "administration_route",
                "dosage",
            ],
            "start_column": "start_date",
            "end_column": "end_date",
            "anchor": anchor,
        }

        sequence_data = pool_data["interval"]

        seqpool = SequencePool.init("interval", sequence_data, settings=settings)
        snapshot.assert_match(seqpool.sequence_data.to_csv())
