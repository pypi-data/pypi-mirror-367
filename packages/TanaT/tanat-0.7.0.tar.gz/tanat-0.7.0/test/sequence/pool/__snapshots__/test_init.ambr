# serializer version: 1
# name: TestInitSequencePool.test_anchoring_interval[end]
  '''
  patient_id,start_date,end_date,medication,administration_route,dosage
  1,2023-01-08,2023-01-18,PAIN_RELIEVER,,2
  1,2023-01-12,2023-01-28,ANTICOAGULANT,CUTANEOUS,1
  1,2023-01-20,2023-02-13,ANTIBIOTIC,ORAL,2
  1,2023-05-23,2023-05-30,ANTIHYPERTENSIVE,INTRAVENOUS,3
  1,2023-09-09,2023-09-20,ANTICOAGULANT,INTRAVENOUS,3
  2,2023-01-05,2023-02-03,ANTIBIOTIC,ORAL,2
  2,2023-01-10,2023-02-09,PAIN_RELIEVER,,1
  2,2023-02-04,2023-02-16,PAIN_RELIEVER,INTRAVENOUS,1
  2,2023-05-25,2023-06-09,ANTIBIOTIC,INHALATION,1
  2,2023-08-08,2023-08-19,ANTIBIOTIC,CUTANEOUS,3
  2,2023-08-08,2023-08-30,ANTIHYPERTENSIVE,INHALATION,2
  3,2023-04-15,2023-04-24,ANTIBIOTIC,INHALATION,3
  3,2023-09-24,2023-10-21,PAIN_RELIEVER,ORAL,2
  3,2023-10-10,2023-11-02,ANTIHYPERTENSIVE,ORAL,2
  4,2023-02-22,2023-03-05,PAIN_RELIEVER,INTRAMUSCULAR,3
  4,2023-05-25,2023-06-21,ANTIBIOTIC,ORAL,3
  4,2023-06-11,2023-07-09,CORTICOSTEROID,INTRAVENOUS,2
  4,2023-06-22,2023-07-11,ANTICOAGULANT,ORAL,1
  4,2023-07-16,2023-07-27,CORTICOSTEROID,INHALATION,3
  4,2023-10-15,2023-10-23,ANTICOAGULANT,INTRAMUSCULAR,2
  5,2023-02-23,2023-03-13,ANTIHYPERTENSIVE,INTRAVENOUS,1
  5,2023-03-07,2023-03-24,CORTICOSTEROID,CUTANEOUS,2
  5,2023-04-22,2023-05-13,ANTIHYPERTENSIVE,INHALATION,2
  5,2023-09-06,2023-09-18,ANTIBIOTIC,CUTANEOUS,2
  5,2023-09-04,2023-09-20,CORTICOSTEROID,,1
  5,2023-10-21,2023-11-14,PAIN_RELIEVER,INTRAMUSCULAR,2
  6,2023-03-10,2023-03-18,ANTIHYPERTENSIVE,INTRAMUSCULAR,1
  6,2023-03-02,2023-03-27,CORTICOSTEROID,INHALATION,1
  6,2023-04-02,2023-04-18,ANTIHYPERTENSIVE,INHALATION,2
  6,2023-04-21,2023-05-20,CORTICOSTEROID,CUTANEOUS,1
  6,2023-05-06,2023-05-23,CORTICOSTEROID,INTRAVENOUS,3
  6,2023-05-18,2023-06-17,ANTIBIOTIC,INTRAMUSCULAR,1
  6,2023-07-14,2023-07-23,CORTICOSTEROID,INHALATION,2
  7,2023-02-03,2023-02-26,PAIN_RELIEVER,INHALATION,3
  7,2023-03-14,2023-04-04,ANTIBIOTIC,,3
  7,2023-05-13,2023-05-28,ANTICOAGULANT,INTRAMUSCULAR,2
  8,2023-01-26,2023-02-05,ANTIBIOTIC,ORAL,3
  8,2023-01-14,2023-02-13,CORTICOSTEROID,,3
  8,2023-01-28,2023-02-13,ANTIBIOTIC,ORAL,3
  8,2023-03-19,2023-04-18,PAIN_RELIEVER,INHALATION,2
  8,2023-08-26,2023-09-12,ANTICOAGULANT,INHALATION,2
  8,2023-09-10,2023-10-06,PAIN_RELIEVER,CUTANEOUS,3
  8,2023-10-21,2023-10-30,CORTICOSTEROID,ORAL,3
  9,2023-04-19,2023-05-13,ANTICOAGULANT,INTRAMUSCULAR,3
  9,2023-05-06,2023-05-24,ANTIBIOTIC,ORAL,1
  9,2023-07-31,2023-08-10,PAIN_RELIEVER,INTRAVENOUS,1
  9,2023-09-21,2023-10-05,ANTIBIOTIC,CUTANEOUS,3
  10,2023-03-11,2023-04-09,ANTICOAGULANT,INTRAVENOUS,2
  10,2023-04-11,2023-04-23,ANTICOAGULANT,INTRAMUSCULAR,2
  10,2023-04-05,2023-05-05,ANTIBIOTIC,INTRAVENOUS,3
  10,2023-08-17,2023-08-24,ANTIHYPERTENSIVE,INHALATION,2
  
  '''
# ---
# name: TestInitSequencePool.test_anchoring_interval[middle]
  '''
  patient_id,start_date,end_date,medication,administration_route,dosage
  1,2023-01-08,2023-01-18,PAIN_RELIEVER,,2
  1,2023-01-12,2023-01-28,ANTICOAGULANT,CUTANEOUS,1
  1,2023-01-20,2023-02-13,ANTIBIOTIC,ORAL,2
  1,2023-05-23,2023-05-30,ANTIHYPERTENSIVE,INTRAVENOUS,3
  1,2023-09-09,2023-09-20,ANTICOAGULANT,INTRAVENOUS,3
  2,2023-01-05,2023-02-03,ANTIBIOTIC,ORAL,2
  2,2023-01-10,2023-02-09,PAIN_RELIEVER,,1
  2,2023-02-04,2023-02-16,PAIN_RELIEVER,INTRAVENOUS,1
  2,2023-05-25,2023-06-09,ANTIBIOTIC,INHALATION,1
  2,2023-08-08,2023-08-19,ANTIBIOTIC,CUTANEOUS,3
  2,2023-08-08,2023-08-30,ANTIHYPERTENSIVE,INHALATION,2
  3,2023-04-15,2023-04-24,ANTIBIOTIC,INHALATION,3
  3,2023-09-24,2023-10-21,PAIN_RELIEVER,ORAL,2
  3,2023-10-10,2023-11-02,ANTIHYPERTENSIVE,ORAL,2
  4,2023-02-22,2023-03-05,PAIN_RELIEVER,INTRAMUSCULAR,3
  4,2023-05-25,2023-06-21,ANTIBIOTIC,ORAL,3
  4,2023-06-11,2023-07-09,CORTICOSTEROID,INTRAVENOUS,2
  4,2023-06-22,2023-07-11,ANTICOAGULANT,ORAL,1
  4,2023-07-16,2023-07-27,CORTICOSTEROID,INHALATION,3
  4,2023-10-15,2023-10-23,ANTICOAGULANT,INTRAMUSCULAR,2
  5,2023-02-23,2023-03-13,ANTIHYPERTENSIVE,INTRAVENOUS,1
  5,2023-03-07,2023-03-24,CORTICOSTEROID,CUTANEOUS,2
  5,2023-04-22,2023-05-13,ANTIHYPERTENSIVE,INHALATION,2
  5,2023-09-04,2023-09-20,CORTICOSTEROID,,1
  5,2023-09-06,2023-09-18,ANTIBIOTIC,CUTANEOUS,2
  5,2023-10-21,2023-11-14,PAIN_RELIEVER,INTRAMUSCULAR,2
  6,2023-03-10,2023-03-18,ANTIHYPERTENSIVE,INTRAMUSCULAR,1
  6,2023-03-02,2023-03-27,CORTICOSTEROID,INHALATION,1
  6,2023-04-02,2023-04-18,ANTIHYPERTENSIVE,INHALATION,2
  6,2023-04-21,2023-05-20,CORTICOSTEROID,CUTANEOUS,1
  6,2023-05-06,2023-05-23,CORTICOSTEROID,INTRAVENOUS,3
  6,2023-05-18,2023-06-17,ANTIBIOTIC,INTRAMUSCULAR,1
  6,2023-07-14,2023-07-23,CORTICOSTEROID,INHALATION,2
  7,2023-02-03,2023-02-26,PAIN_RELIEVER,INHALATION,3
  7,2023-03-14,2023-04-04,ANTIBIOTIC,,3
  7,2023-05-13,2023-05-28,ANTICOAGULANT,INTRAMUSCULAR,2
  8,2023-01-14,2023-02-13,CORTICOSTEROID,,3
  8,2023-01-26,2023-02-05,ANTIBIOTIC,ORAL,3
  8,2023-01-28,2023-02-13,ANTIBIOTIC,ORAL,3
  8,2023-03-19,2023-04-18,PAIN_RELIEVER,INHALATION,2
  8,2023-08-26,2023-09-12,ANTICOAGULANT,INHALATION,2
  8,2023-09-10,2023-10-06,PAIN_RELIEVER,CUTANEOUS,3
  8,2023-10-21,2023-10-30,CORTICOSTEROID,ORAL,3
  9,2023-04-19,2023-05-13,ANTICOAGULANT,INTRAMUSCULAR,3
  9,2023-05-06,2023-05-24,ANTIBIOTIC,ORAL,1
  9,2023-07-31,2023-08-10,PAIN_RELIEVER,INTRAVENOUS,1
  9,2023-09-21,2023-10-05,ANTIBIOTIC,CUTANEOUS,3
  10,2023-03-11,2023-04-09,ANTICOAGULANT,INTRAVENOUS,2
  10,2023-04-11,2023-04-23,ANTICOAGULANT,INTRAMUSCULAR,2
  10,2023-04-05,2023-05-05,ANTIBIOTIC,INTRAVENOUS,3
  10,2023-08-17,2023-08-24,ANTIHYPERTENSIVE,INHALATION,2
  
  '''
# ---
# name: TestInitSequencePool.test_anchoring_interval[start]
  '''
  patient_id,start_date,end_date,medication,administration_route,dosage
  1,2023-01-08,2023-01-18,PAIN_RELIEVER,,2
  1,2023-01-12,2023-01-28,ANTICOAGULANT,CUTANEOUS,1
  1,2023-01-20,2023-02-13,ANTIBIOTIC,ORAL,2
  1,2023-05-23,2023-05-30,ANTIHYPERTENSIVE,INTRAVENOUS,3
  1,2023-09-09,2023-09-20,ANTICOAGULANT,INTRAVENOUS,3
  2,2023-01-05,2023-02-03,ANTIBIOTIC,ORAL,2
  2,2023-01-10,2023-02-09,PAIN_RELIEVER,,1
  2,2023-02-04,2023-02-16,PAIN_RELIEVER,INTRAVENOUS,1
  2,2023-05-25,2023-06-09,ANTIBIOTIC,INHALATION,1
  2,2023-08-08,2023-08-19,ANTIBIOTIC,CUTANEOUS,3
  2,2023-08-08,2023-08-30,ANTIHYPERTENSIVE,INHALATION,2
  3,2023-04-15,2023-04-24,ANTIBIOTIC,INHALATION,3
  3,2023-09-24,2023-10-21,PAIN_RELIEVER,ORAL,2
  3,2023-10-10,2023-11-02,ANTIHYPERTENSIVE,ORAL,2
  4,2023-02-22,2023-03-05,PAIN_RELIEVER,INTRAMUSCULAR,3
  4,2023-05-25,2023-06-21,ANTIBIOTIC,ORAL,3
  4,2023-06-11,2023-07-09,CORTICOSTEROID,INTRAVENOUS,2
  4,2023-06-22,2023-07-11,ANTICOAGULANT,ORAL,1
  4,2023-07-16,2023-07-27,CORTICOSTEROID,INHALATION,3
  4,2023-10-15,2023-10-23,ANTICOAGULANT,INTRAMUSCULAR,2
  5,2023-02-23,2023-03-13,ANTIHYPERTENSIVE,INTRAVENOUS,1
  5,2023-03-07,2023-03-24,CORTICOSTEROID,CUTANEOUS,2
  5,2023-04-22,2023-05-13,ANTIHYPERTENSIVE,INHALATION,2
  5,2023-09-04,2023-09-20,CORTICOSTEROID,,1
  5,2023-09-06,2023-09-18,ANTIBIOTIC,CUTANEOUS,2
  5,2023-10-21,2023-11-14,PAIN_RELIEVER,INTRAMUSCULAR,2
  6,2023-03-02,2023-03-27,CORTICOSTEROID,INHALATION,1
  6,2023-03-10,2023-03-18,ANTIHYPERTENSIVE,INTRAMUSCULAR,1
  6,2023-04-02,2023-04-18,ANTIHYPERTENSIVE,INHALATION,2
  6,2023-04-21,2023-05-20,CORTICOSTEROID,CUTANEOUS,1
  6,2023-05-06,2023-05-23,CORTICOSTEROID,INTRAVENOUS,3
  6,2023-05-18,2023-06-17,ANTIBIOTIC,INTRAMUSCULAR,1
  6,2023-07-14,2023-07-23,CORTICOSTEROID,INHALATION,2
  7,2023-02-03,2023-02-26,PAIN_RELIEVER,INHALATION,3
  7,2023-03-14,2023-04-04,ANTIBIOTIC,,3
  7,2023-05-13,2023-05-28,ANTICOAGULANT,INTRAMUSCULAR,2
  8,2023-01-14,2023-02-13,CORTICOSTEROID,,3
  8,2023-01-26,2023-02-05,ANTIBIOTIC,ORAL,3
  8,2023-01-28,2023-02-13,ANTIBIOTIC,ORAL,3
  8,2023-03-19,2023-04-18,PAIN_RELIEVER,INHALATION,2
  8,2023-08-26,2023-09-12,ANTICOAGULANT,INHALATION,2
  8,2023-09-10,2023-10-06,PAIN_RELIEVER,CUTANEOUS,3
  8,2023-10-21,2023-10-30,CORTICOSTEROID,ORAL,3
  9,2023-04-19,2023-05-13,ANTICOAGULANT,INTRAMUSCULAR,3
  9,2023-05-06,2023-05-24,ANTIBIOTIC,ORAL,1
  9,2023-07-31,2023-08-10,PAIN_RELIEVER,INTRAVENOUS,1
  9,2023-09-21,2023-10-05,ANTIBIOTIC,CUTANEOUS,3
  10,2023-03-11,2023-04-09,ANTICOAGULANT,INTRAVENOUS,2
  10,2023-04-05,2023-05-05,ANTIBIOTIC,INTRAVENOUS,3
  10,2023-04-11,2023-04-23,ANTICOAGULANT,INTRAMUSCULAR,2
  10,2023-08-17,2023-08-24,ANTIHYPERTENSIVE,INHALATION,2
  
  '''
# ---
